### PR TITLE
Fix broken tests in Datetime Parser

### PR DESCRIPTION
**Pull request by hishamco**
Related to #10601

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DateTimeParser.cs [hunk 1]
```diff
@@ -67,22 +67,6 @@ public override string ToString()
             => $"{(String.IsNullOrEmpty(Operator) ? String.Empty : Operator)}{DateTime.ToString("o")}";
     }
 
-    public class DateNode2 : OperatorNode
```
Comment (deanmarcussen) [inline src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DateTimeParser.cs:70]: I would have to understand why I added this, it was for a reason. So I disagree with removing it.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DateTimeParser.cs:70]: > So I disagree with removing it.

Take your time to check this, I notice that the `DateNode` that you added is based on `DateTime` which makes issues with offset, now I need to asked you what are the reasons for adding this ;)

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DateTimeParser.cs:70]: This method implies that the date passed in a UtcDateTime. Just add comments on the method.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DateTimeParser.cs:70]: @Skrypt can you elaborate?

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DateTimeParser.cs:70]: What I'm saying is that if DateNode2 uses a DateTime as a parameter and that this DateTime should always be as UTC then you can also validate it inside the DateTime2 constructor instead to keep what we had. If we remove DateNode2 then we need to keep it and mark it as Obsolete instead. But, I assume here that this is used only by the AuditTrail so the changes you made are fine. So, up to @deanmarcussen to illustrate his point.

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/DateTimeParser.cs:70]: I'm still waiting for @deanmarcussen to clarify why `DateTime2` is here, it's fine to mark it as `obsolete` for now

**Comment by hishamco** [comment]
@Skrypt can we revise this PR while Dean is absent for awhile

**Comment by Piedone** [comment]
I checked this out but really have nothing to add.

**Comment by Skrypt** [comment]
You will need to explain to me what this piece of code does because I have no idea as of now. I would need to analyze the Audit Trail module to understand it better. But here, we have 2 classes that have either a DatetimeOffset or a Datetime as a parameter to differentiate them. Normally, you should use a different constructor on the same class for that matter. But at the same time, the information that no one gave me is about if it is necessary to have a UTC DateTime here. Because then, the class that has a DateTime parameter would need to make sure that the DateTime passed is of UtcKind.

https://docs.microsoft.com/en-us/dotnet/api/system.datetimekind?view=net-6.0

Then, why there is 2 different classes for doing this is what needs to be defined before anything else.

**Comment by hishamco** [comment]
> Then, why there are 2 different classes for doing this is what needs to be defined before anything else.

That's why I waiting for the Dean from awhile, please check his comment [here](https://github.com/OrchardCMS/OrchardCore/pull/10645#discussion_r744108677)

**Comment by deanmarcussen** [comment]
> > Then, why there are 2 different classes for doing this is what needs to be defined before anything else.
> 
> That's why I waiting for the Dean from awhile, please check his comment [here](https://github.com/OrchardCMS/OrchardCore/pull/10645#discussion_r744108677)

I made it quite clear this was not a good pr. Please do not remove the second date class, it is there for a reason.

**Comment by hishamco** [comment]
I'm stilll waiting for that reason ;) we could check where is it used or fix uncommented tests in the way you prefer

**Comment by Skrypt** [comment]
Unless it is an issue for someone "which I would really be surprised" then I suggest we leave it as it is for now and come back later if it becomes a need. Else, we are just optimizing code for optimizing it while potentially breaking implementations.

**Comment by Piedone** [comment]
You can rename `DateNode` to `DateTimeOffsetNode` and `DateNode2` to e.g. `DateTimeNode` if you're really inclined to, but that's about what I think would be useful here, apart from fixing the test.

**Comment by Piedone** [comment]
Superseded by https://github.com/OrchardCMS/OrchardCore/pull/15297.

**Pull request closed without merging**